gips/data/landsat/landsat.py

```diff
@@ -1025,49 +994,36 @@ def _process_indices(self, image, asset_fn, metadata, sensor, indices,
         is a dict with keys `x` and `y` used to make affine
         transformation for `-coreg` products.
         """
-        gippy_input = {} # map prod types to temp output filenames for feeding to gippy
-        tempfps_to_ptypes = {} # map temp output filenames to prod types, for AddFile
-        for prod_type, pt_split in indices.items():
-            temp_fp = self.temp_product_filename(sensor, prod_type)
-            gippy_input[pt_split[0]] = temp_fp
-            tempfps_to_ptypes[temp_fp] = prod_type
-
-        self._time_report("Running Indices")
-        prodout = Indices(image, gippy_input,
-                          self.prep_meta(asset_fn, metadata))
-        self._time_report("Finshed running Indices")
-
-        if coreg_shift:
-            for key, val in prodout.iteritems():
+        verbose_out("Starting on {} indices: {}".format(len(indices), indices.keys()), 2)
+        for prod_and_args, split_p_and_a in indices.items():
```

Comment:
TODO: some sort of gippy.algorithms.indices optimization because there are extra reads in this case (unless some gdal caching is happening).  Test and document difference in runtime of  producing 3 indices in gippy0.3.12 vs gippy1.0.  Some sort of metric of how big of a hit are the redundant reads.

Reply:
internal issue filed